conbench/entities/history.py

```diff
@@ -255,58 +254,63 @@ def get_history_for_cchr(
     return samples
 
 
-def set_z_scores(benchmark_results: List[BenchmarkResult]):
+def set_z_scores(benchmark_results: List[BenchmarkResult], baseline_commit: Commit):
     """Set the "z_score" attribute on each given BenchmarkResult, comparing it to the
-    distribution of BenchmarkResults that share the same case/context/hardware/repo,
-    ending with the most recent BenchmarkResult in the given BenchmarkResult's commit
-    ancestry on the default branch. (We choose a commit from the default branch as the
-    end commit, because we probably don't want "bad" commits early on in a PR to set
-    artificially low z-scores later in the PR.)
+    distribution of BenchmarkResults that share the same case/context/hardware/repo, in
+    the git ancestry of the baseline_commit (inclusive).
+
+    The given benchmark_results must all have the same run_id.
```

Comment:
It's good to call out invariants like this in the docstring. I see that below we make that count with the typical set() approach and otherwise raise a ValueError. Good!